cmd/clusterctl/cmd/config_repositories.go

```diff
@@ -40,21 +47,26 @@ var configRepositoryCmd = &cobra.Command{
 
 	Example: Examples(`
 		# Displays the list of available providers.
-		clusterctl config repositories`),
+		clusterctl config repositories
+		
+		# Print the list of available providers in yaml format.
+		clusterctl config repositories -o yaml`),
 
 	RunE: func(cmd *cobra.Command, args []string) error {
 		return runGetRepositories(cfgFile, os.Stdout)
 	},
 }
 
 func init() {
+	configRepositoryCmd.Flags().StringVarP(&cro.output, "output", "o", "", "Output format; available options are 'yaml'")
```

Comment:
For sake of consistency I suggest implementing something similar to what exists in `clusterctl config provider`:
- Available output formats are defined as a const and as a list
- Available formats should include text, which is the default format